### PR TITLE
Application adapter fix [OSF-6698]

### PR DESCRIPTION
**Pull request by jnayak1**
Switches to using JSONAPIAdapter instead of OsfAdapter. Since we are not making requests from ember to OSF directly, there is no need to use OsfAdapter. Fixes CSRF issues by placing CSRF token in header.

### CODE REVIEW

File: app/adapters/application.js [hunk 1]
```diff
@@ -1,5 +1,29 @@
-import OsfAdapter from './osf-adapter';
+import DS from 'ember-data';
+export default DS.JSONAPIAdapter.extend({
+    host: 'http://localhost:8000',
+    buildURL(modelName, id, snapshot, requestType) {
```
Comment (Ghalko) [inline app/adapters/application.js:4]: We really need to make this (localhost:8000) part of the config, otherwise it is going to be really hard to deploy.

Comment (Ghalko) [inline app/adapters/application.js:4]: But I'll make a ticket for it.

File: app/adapters/application.js [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+		hash.crossDomain = true;
+		hash.xhrFields = {withCredentials: true};
+		return this._super(url, method, hash);
+  	},
+  	headers: Ember.computed(function() {
+  		var csrftoken = ""
+  		try {
+  			var csrftoken = Ember.get(document.cookie.match(/csrftoken\=([^;]*)/), "1")
+  		} catch(e){
+  			console.log(e)
```
Comment (Ghalko) [inline app/adapters/application.js:28]: Maybe use an ember error.

**Comment by jnayak1** [comment]
Oh yes, we have done this in many places

File: app/adapters/application.js (hunk 2 above)
Comment (jnayak1) [inline app/adapters/application.js]: Oh yes, we have done this in too many places

**Comment by Ghalko** [comment]
Which one? Both?

**Comment by jnayak1** [comment]
I meant the localhost:8000. The Ember Error class looks empty on the Ember Api. http://emberjs.com/api/classes/Ember.Error.html

**Comment by Ghalko** [comment]
http://emberjs.com/api/data/classes/DS.Errors.html

**Comment by jnayak1** [comment]
Ah! Yes I'll fix that.

**Comment by Ghalko** [comment]
Or using toast for bubbling up server errors.

**Comment by jnayak1** [comment]
> http://emberjs.com/api/data/classes/DS.Errors.html

Yeah it appears this is for Ember Data, not Ember.

**Pull request merged**